pyocd/coresight/generic_mem_ap.py

```diff
@@ -0,0 +1,153 @@
+# pyOCD debugger
+# Copyright (c) 2020 Cypress Semiconductor Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from .component import CoreSightCoreComponent
+from ..core.target import Target
+
+LOG = logging.getLogger(__name__)
+
+DEAD_VALUE = 0
+
+
+class GenericMemAP(Target, CoreSightCoreComponent):
```

Comment:
Can we rename this class to `GenericMemAPTarget`?

Also, please add at least a one line doc comment for the class, doxygen style, e.g.`"""! @brief foo"""`.

Fyi: I'm thinking about adding a "target capabilities" set so the target can specify, for instance, whether breakpoints are supported.